examples/platform/silabs/provision/ProvisionStorageDefault.cpp

```diff
@@ -70,14 +70,36 @@ CHIP_ERROR ErasePage(uint32_t addr)
     return chip::DeviceLayer::Silabs::GetPlatform().FlashErasePage(addr);
 }
 
-CHIP_ERROR WritePage(uint32_t addr, const uint8_t * data, size_t size)
+size_t RoundNearest(size_t n, size_t multiple)
 {
-    return chip::DeviceLayer::Silabs::GetPlatform().FlashWritePage(addr, data, size);
+    return (n % multiple) > 0 ? n + (multiple - n % multiple) : n;
 }
 
-size_t RoundNearest(size_t n, size_t multiple)
+/**
+ * Writes "size" bytes to the flash page. The data is padded with 0xff
+ * up to the nearest 32-bit boundary. The "max" argument is used to ensure
+ * that the padding won't exceed the limits of the buffer.
+ */
+CHIP_ERROR WritePage(uint32_t addr, const uint8_t * data, size_t size)
 {
-    return (n % multiple) > 0 ? n + (multiple - n % multiple) : n;
+    // The flash driver fails if the size is not a multiple of 4 (32-bits)
+    size_t size_32 = RoundNearest(size, 4);
+    if (size_32 == size)
+    {
+        // The given data is already aligned to 32-bit
+        return chip::DeviceLayer::Silabs::GetPlatform().FlashWritePage(addr, data, size);
+    }
+    else
+    {
+        // Create a temporary buffer, and pad it with "0xff"
+        uint8_t * p = new uint8_t[size_32];
+        VerifyOrReturnError(p != nullptr, CHIP_ERROR_INTERNAL);
```

Comment:
As far as I know, `new` without std::nothrow will never return null: it will throw an exception (or crash if exceptions are disabled), no?

Reply:
You're correct. I have replaced the new operator with Platform::MemoryAlloc().